### DISCUSSION TITLE
Slow start and freeze web interface.

### DISCUSSION THREAD

**Discussion by ur5fan** [PhantomSDR/PhantomSDR#23] [Q&A] [answered]
Hello. I'm playing with the PhantomSdr and RX888 mk2. Here is my start command: `sudo ./rx888_stream -f ../rx888_stream/SDDC_FX3.img  -r --sample-rate 10000000 -o - | build/spectrumserver --config config.toml`
And my config.toml
```
[server]
port=9002 # Server port
html_root="html-svelte/dist/" # HTML files to be hosted
otherusers=1 # Send where other users are listening, 0 to disable
threads=1

[register] # Register with the server
enable=false # Set to true to publish the server
name="Name"
antenna="Antenna"
description="Description"
url="" # Custom URL, if not server assigns http://ip:port
password="" # Secret ID/password to register, ensures only you can update the details, if left blank a random ID is generated

[limits]
audio=1000
waterfall=1000
events=1000

[input]
sps=10000000 # Input Sample Rate
fft_size=8388608 # FFT bins
fft_threads=4 # FFT threads
frequency=1500000 # Baseband frequency
signal="real" # real or iq
accelerator="none" # Accelerator: none, cuda, opencl
audio_sps=48000 # Audio Sample Rate
audio_compression="flac" # flac or opus
waterfall_size=2048
waterfall_compression="zstd" # zstd or av1

[input.driver]
name="stdin" # Driver name
format="s16" # Sample format: u8, s8, u16, s16, u32, s32, f32, f64

[input.defaults]
frequency=15000000 # Default frequency to show user
modulation="AM" # Default modulation
```
My laptop
```
 Linux mint(Debian)
 Chrome browser
 Intel Core i7-2640M 4 cors
 Memory: 7.64 GiB
 Intel 2nd Generation Core Processor Family Integrated Graphics
```
 I need > 1 minute (sometimes 3-4 minutes) until a spectrum and audio appear in my web browser. 
 After a couple of minutes, it becomes more stable. Is such behavior expected? What are the requirements for the server side?
 
![Screenshot from 2024-06-16 16-57-42](https://github.com/PhantomSDR/PhantomSDR/assets/4150389/5528ce1a-9dcd-429f-8fd2-4136c3ae7910)

![Screenshot from 2024-06-16 16-58-22](https://github.com/PhantomSDR/PhantomSDR/assets/4150389/5b339c62-a8e0-4309-8794-e3ba0c5e5e6c)

**Comment by rhgndf** [comment, marked as answer]
This is normal, since FFTW needs to do some benchmark before the FFT starts up. 

To get rid of it change FFTW_MEASURE to FFTW_ESTIMATE in the lines here: https://github.com/PhantomSDR/PhantomSDR/blob/b9951cbc3f5ff20854a3616a9edb7117b5ecd3f5/src/fft.cpp#L23

That might reduce the performance due to a suboptimal FFT implementation being selected.

**Reply by ur5fan** [reply]
Thanks a lot!
